src/MongoDB.Bson/Serialization/Serializers/DiscriminatedInterfaceSerializer.cs

```diff
@@ -110,5 +110,25 @@ public override void Serialize(BsonSerializationContext context, BsonSerializati
                 _objectSerializer.Serialize(context, args, value);
             }
         }
+
+        /// <inheritdoc />
+        public bool TryGetMemberSerializationInfo(string memberName, out BsonSerializationInfo serializationInfo)
+        {
+            var memberType = _interfaceType.GetTypeInfo().GetProperty(memberName);
+            if (memberType != null)
+            {
+                var serializer = BsonSerializer.LookupSerializer(memberType.PropertyType);
+                serializationInfo = new BsonSerializationInfo(
+                    memberName,
+                    serializer,
+                    typeof(TInterface));
```

Comment:
Shouldn't this be the type of the member (not the type of the interface)?

Reply:
sgtm